src/main/java/ua/softserveinc/tc/dao/impl/TokenDaoImpl.java

```diff
@@ -9,40 +9,40 @@
 import ua.softserveinc.tc.entity.User;
 import ua.softserveinc.tc.util.Log;
 
-import javax.persistence.NoResultException;
+import javax.persistence.EntityManager;
+import javax.persistence.PersistenceContext;
 import javax.persistence.Query;
+import javax.persistence.TypedQuery;
 
-
-/**
- * Created by Chak on 17.05.2016.
- */
 @Repository
 public class TokenDaoImpl extends BaseDaoImpl<Token> implements TokenDao {
 
     @Log
     private static Logger log;
 
+    @PersistenceContext
+    private EntityManager entityManager;
+
     @Override
     public void delete(Token entity) {
-        super.delete(getEntityManager().merge(entity));
+        super.delete(entityManager.merge(entity));
     }
 
     @Override
     public Token findByUser(User user) {
-        Query query = getEntityManager().createQuery("from Token where user.id = :user");
+        Query query = entityManager.createQuery("from Token where user.id = :user");
         query.setParameter(UserConstants.Entity.USER, user.getId());
         return (Token) query.getSingleResult();
     }
 
     @Override
     public Token findByToken(String token) {
-        try {
-            Query query = getEntityManager().createQuery("from Token where token = :token");
-            return (Token) query.setParameter(TokenConstants.TOKEN, token).getSingleResult();
-        } catch (NoResultException e) {
-            log.error("no exist", e);
-            return null;
```

Comment:
Using exception in control flow is a [bad practise](http://programmers.stackexchange.com/questions/189222/are-exceptions-as-control-flow-considered-a-serious-antipattern-if-so-why).